packages/rulesets/src/oas/functions/oasPathParam.ts

```diff
@@ -2,7 +2,7 @@ import type { JsonPath, Segment } from '@stoplight/types';
 import type { IFunction, IFunctionResult } from '@stoplight/spectral-core';
 import { isObject } from './utils/isObject';
 
-const pathRegex = /(\{[a-zA-Z0-9_-]+\})+/g;
+const pathRegex = /(\{;?\??[a-zA-Z0-9_-]+\*?\})/g;
```

Comment:
I believe we should be a bit more specific, and apply this regular expression only when `style=matrix`.
What do you think?

Reply:
regarding to the docs (https://swagger.io/docs/specification/serialization/#path), it can always end with `*`  when `explode=true`, and can begin with ``, `.`, `;` depending on style=`simple`, `label`, `matrix`. also in my understanding it can begin with `?` or `&` for style= `form` / `pipeDelimited` / `spaceDelimited`. and moreover, can begin with `+` for `allowReserved=true`. 

So following the whole specs would need a lot of changes and edge case checks. Question here if its more okay-ish to get errors on good path's, or to not get errors on bad path's.